staging/src/k8s.io/code-generator/cmd/client-gen/generators/generator_for_type.go

```diff
@@ -206,51 +210,49 @@ func (g *genClientForType) GenerateType(c *generator.Context, t *types.Type, w i
 	}
 	sw.Do(interfaceTemplate4, m)
 
-	if tags.NonNamespaced {
-		sw.Do(structNonNamespaced, m)
-		sw.Do(newStructNonNamespaced, m)
-	} else {
-		sw.Do(structNamespaced, m)
-		sw.Do(newStructNamespaced, m)
-	}
-
 	if tags.NoVerbs {
-		return sw.Error()
-	}
+		sw.Do(structType, m)
+		if tags.NonNamespaced {
+			sw.Do(newStructNonNamespaced, m)
+		} else {
+			sw.Do(newStructNamespaced, m)
+		}
 
-	if tags.HasVerb("get") {
```

Comment:
I’ve reworked this to use a truth-table-style index into an array of templates, the code seems much clearer that way. (It could be made even clearer if `HasVerb` did a `NoVerb` check itself, but I haven’t looked into the ramifications of that yet.)